AirCasting/Map/GoogleMap/GoogleMapView.swift

```diff
@@ -0,0 +1,184 @@
+    
+    @Binding private var noteMarketTapped: Bool
+    @Binding private var noteNumber: Int
+    
+    private var liveModeOn: Bool
+    private var isSessionFixed: Bool
+    private var pathPoints: [PathPoint]
+    
+    private(set) var threshold: SensorThreshold?
+    private var onPositionChange: (([PathPoint]) -> ())? = nil
+    
+    init(pathPoints: [PathPoint],
+         threshold: SensorThreshold? = nil,
+         isMyLocationEnabled: Bool = false,
+         isSessionActive: Bool = false,
+         isSessionFixed: Bool = false,
+         noteMarketTapped: Binding<Bool> = .constant(false),
+         noteNumber: Binding<Int> = .constant(0),
+         mapNotes: Binding<[MapNote]> = .constant([]),
+         showMyLocationButton: Bool = true) {
+        
+        self.pathPoints = pathPoints
+        self.threshold = threshold
+        self.liveModeOn = isSessionActive
+        self.isSessionFixed = isSessionFixed
+        self._noteMarketTapped = noteMarketTapped
+        self._noteNumber = noteNumber
+        self._butler = .init(wrappedValue: .init(pathPoints: pathPoints,
+                                                 showMyLocationButton: showMyLocationButton,
+                                                 isMyLocationEnabled: isMyLocationEnabled,
+                                                 isSessionActive: isSessionActive,
+                                                 isSessionFixed: isSessionFixed,
+                                                 threshold: threshold,
+                                                 notes: mapNotes))
+    }
+    
+    /// Adds an action for when the map viewport is changed.
+    /// - Parameter action: an action block that takes an array of currently visible `PathPoint`s.
+    func onPositionChange(action: @escaping (_ visiblePoints: [PathPoint]) -> ()) -> Self {
+        var newSelf = self
+        newSelf.onPositionChange = action
+        return newSelf
+    }
+    
+    func makeUIView(context: Context) -> GMSMapView {
+        let mapView = butler.initialMakeUIVIew(context: context)
+        
+        mapView.delegate = context.coordinator
+        context.coordinator.mapNotesCounter = butler.notes.count
+        context.coordinator.myLocationSink = mapView.publisher(for: \.myLocation)
+            .sink { [weak mapView] (location) in
+                guard let coordinate = location?.coordinate else { return }
+                mapView?.animate(toLocation: coordinate)
+            }
+        updateContextThresholdAndPathPoints(context: context)
+        
+        DispatchQueue.main.async {
+            mapView.moveCamera(butler.cameraUpdate)
+            if mapView.camera.zoom > 16 {  mapView.animate(toZoom: 16) }
+            // Centers the user onto last recorded position
+        }
+        
+        return mapView
+    }
+    
+    func updateUIView(_ uiView: GMSMapView, context: Context) {
+        context.coordinator.butler = butler
```

Comment:
Why do we need to assign butler on updateUIView?

Reply:
To keep it up to date and be sure that data transfers on every map update - this is why. I will try to explain it further in the upcoming days as I will gain more knowledge which I would like to get first.